src/states/infiniteweightpeps.jl

```diff
@@ -0,0 +1,211 @@
+
+"""
+    const PEPSWeight{S}
+
+Default type for PEPS bond weights with 2 virtual indices,
+conventionally ordered as: ``wt : ES ← WN``. 
+Here, `ES`, `WN` denote the east/south, west/north spaces, respectively.
+"""
+const PEPSWeight{S} = AbstractTensorMap{S,1,1} where {S<:ElementarySpace}
+
+"""
+Schmidt bond weight used in simple/cluster update
+"""
+struct SUWeight{E<:PEPSWeight}
+    x::Matrix{E}
+    y::Matrix{E}
+end
+
+function Base.size(wts::SUWeight)
+    @assert size(wts.x) == size(wts.y)
```

Comment:
I think if you move this to check to the constructor, it is fair to remove it here.